### PR TITLE
Update XamlX

### PR DESCRIPTION
**Pull request by MrJul**
## What does the pull request do?
Updates XamlX to the latest master.
Avalonia's code is adjusted to use the new `XamlVisibility` enum.

## Remarks
`DefineMethod` has a different behavior than before with `XamlVisibility.Private`.

Before, with `isPublic = false`, the method was generated _without any access modifier_, which maps to the "compiler controlled" access modified:
>  compiler-controlled – accessible only through the use of a definition, not a reference, hence only accessible from within a single compilation unit and under the control of the compiler. 

(ECMA-335 6th edition, section I.8.5.3.2)

Basically it acts a bit like C# `internal` since the compilation unit is the whole assembly here, but even stricter (can't be accessed even with `InternalsVisibleTo`). That's why `DefineBuildMethod` now uses `Assembly`, since the `Build` method for a class can be called through the `Populate` method of another class: it was the only case I could find that was taking advantage (involuntarily?) of the compiler-controlled visibility.

This behavior better matches what we would have written in C#, but it's still a change in behavior.

### CODE REVIEW

**Review by maxkatz6 — APPROVED** [review]

**Comment by MrJul** [comment]
The API compat tool reported errors in `Avalonia.Designer.AppHost` due to the changes in XamlX.

The AppHost is a tool and shouldn't normally be referenced by user code: `XamlX` has been made internal in this project (the Avalonia compiler extensions to XamlX already were), and API diff suppressions have been added.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0040546-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]
